### PR TITLE
[serve] Improve DEBUG logging for understanding perf

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Adds a bunch of debug statements to help debug request latency.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by architkulkarni** [comment]
Thanks! I'll take over this PR and make sure tests pass

**Comment by edoakes** [comment]
Thanks @architkulkarni. Btw, it would also be great if we had tests that explicitly set SERVE_LOG_DEBUG (or whatever it is) to make sure those log messages don't error.

**Comment by architkulkarni** [comment]
Good idea.  I uncommented a line in `conftest.py` that that turns on the debug log for all tests--is there a downside to this?

**Review by simon-mo — APPROVED** [review]
can we use context manager per these kind of perf measurement? that should pave way for distributed tracing. but we can also do it for next PR!

File: python/ray/serve/handle.py [hunk 1]
```diff
@@ -60,6 +61,7 @@ def remote(self, request_data: Optional[Union[Dict, Any]] = None,
                 ``request.args``.
         """
         request_metadata = RequestMetadata(
+            get_random_letters(10),  # Used for debugging.
```
[the remark this answers is not in the document]
Comment (edoakes) [reply python/ray/serve/handle.py:64]: Good question. We should run the microbenchmark @architkulkarni. If it's too slow we can do something faster (e.g., use an int).

[the remark this answers is not in the document]
Comment (architkulkarni) [reply python/ray/serve/handle.py:64]: Looks like there's no impact: https://gist.github.com/architkulkarni/5bbe232d4a944584ba4c7b1bbe0f650d

[the remark this answers is not in the document]
Comment (simon-mo) [reply python/ray/serve/handle.py:64]: Can you try post the result of just benching this in ipython?
```
$ ipython
>>> ... import get_random_letters
>>> %timeit get_random_letters(10)
```

[the remark this answers is not in the document]
Comment (architkulkarni) [reply python/ray/serve/handle.py:64]:
```
>>> %timeit get_random_letters(10)                                             
2.62 µs ± 35.4 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

**Pull request merged**